### PR TITLE
plugins/lazygit: init

### LINKED ISSUE #1468: [PLUGIN REQUEST] lazygit.nvim
| Field    | Description |
|----------|-------------|
| Plugin   | `lazygit.nvim`  |
| Homepage | [`kdheepak/lazygit.nvim`](https://github.com/kdheepak/lazygit.nvim)     |
| Nixpkgs  | [`nixpkgs/lazygit`](https://github.com/NixOS/nixpkgs/blob/nixos-23.11/pkgs/development/tools/lazygit/default.nix#L24) |

## Extra Information
A really nice AIO git tool, integration in Nvim is really nice to have.

### PR DESCRIPTION
**Pull request by theabm**
This PR adds the [lazygit.nvim](https://github.com/kdheepak/lazygit.nvim) plugin. 

closes #1468.

### CODE REVIEW

File: plugins/git/lazygit.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      config_file_path =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            either
+            str (listOf str)
+        )
+        "" "Custom config file path or list of custom config file paths.";
+    };
+    settingsExample = {
```
Comment (GaetanLepage) [inline plugins/git/lazygit.nix:56]:
```suggestion
    };

    settingsExample = {
```

File: plugins/git/lazygit.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      floating_window_winblend = 0;
+      floating_window_scaling_factor = 0.9;
+      floating_window_border_chars = ["╭" "─" "╮" "│" "╯" "─" "╰" "│"];
+      floating_window_use_plenary = false;
+      use_neovim_remote = true;
+      use_custom_config_file_path = false;
+      config_file_path = "";
+    };
```
Comment (GaetanLepage) [inline plugins/git/lazygit.nix:64]:
```suggestion
    };
    
    extraOptions = {
      gitPackage = mkOption {
        type = with types; nullOr package;
        default = pkgs.git;
        example = null;
        description = ''
          The `git` package to use.
          Set to `null` to not install any package.
        '';
      };
      
      lazygitPackage = mkOption {
        type = with types; nullOr package;
        default = pkgs.lazygit;
        example = null;
        description = ''
          The `lazygit` package to use.
          Set to `null` to not install any package.
        '';
      };
    };
    
    extraConfig = cfg: {
      extraPackages = [
        cfg.gitPackage
        cfg.lazygitPackage
      ];
    };

```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: plugins/git/lazygit.nix (hunk 1 above)
Comment (theabm) [reply plugins/git/lazygit.nix:56]: done

File: plugins/git/lazygit.nix [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      floating_window_scaling_factor =
+        helpers.defaultNullOpts.mkNullable types.numbers.nonnegative
+        "0.9" "Set the scaling factor for floating window.";
+
+      floating_window_border_chars =
+        helpers.defaultNullOpts.mkListOf types.str
+        ''["╭" "─" "╮" "│" "╯" "─" "╰" "│"]''
+        ''
+          Customize lazygit popup window border characters.
+        '';
```
Comment (GaetanLepage) [inline plugins/git/lazygit.nix:33]:
```suggestion
        "Customize lazygit popup window border characters.";
```

File: plugins/git/lazygit.nix (hunk 1 above)
Comment (GaetanLepage) [inline plugins/git/lazygit.nix:56]:
```suggestion
    };

    settingsExample = {
```

File: plugins/git/lazygit.nix (hunk 2 above)
Comment (theabm) [reply plugins/git/lazygit.nix:64]: done

File: plugins/git/lazygit.nix (hunk 3 above)
Comment (theabm) [reply plugins/git/lazygit.nix:33]: done

File: plugins/git/lazygit.nix [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    maintainers = [helpers.maintainers.AndresBermeoMarinelli];
+
+    settingsOptions = {
+      floating_window_winblend = helpers.defaultNullOpts.mkNullable (types.ints.between 0 100) "0" ''
+        Set the transparency of the floating window.
+      '';
+
+      floating_window_scaling_factor =
+        helpers.defaultNullOpts.mkNullable types.numbers.nonnegative
+        "0.9" "Set the scaling factor for floating window.";
```
Comment (GaetanLepage) [inline plugins/git/lazygit.nix:26]:
```suggestion
      floating_window_scaling_factor = helpers.defaultNullOpts.mkNullable (types.numbers.between 0.0 1.0) "0.9" ''
        Set the scaling factor for floating window.
      '';
```

Comment (GaetanLepage) [reply plugins/git/lazygit.nix:26]: Maybe I'm wrong on this one. Can it be > 1 ?

File: plugins/git/lazygit.nix [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      '';
+
+      config_file_path =
+        helpers.defaultNullOpts.mkNullable
+        (
+          with types;
+            either
+            str (listOf str)
+        )
+        "" "Custom config file path or list of custom config file paths.";
```
Comment (GaetanLepage) [inline plugins/git/lazygit.nix:52]:
```suggestion
        "[]"
        "Custom config file path or list of custom config file paths.";
```

File: tests/test-sources/plugins/git/lazygit.nix [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      enable = true;
+
+      settings = {
+        floating_window_winblend = 0;
+        floating_window_scaling_factor = 0.9;
+        floating_window_border_chars = ["╭" "─" "╮" "│" "╯" "─" "╰" "│"];
+        floating_window_use_plenary = false;
+        use_neovim_remote = true;
+        use_custom_config_file_path = false;
+        config_file_path = "";
```
Comment (GaetanLepage) [inline tests/test-sources/plugins/git/lazygit.nix:17]:
```suggestion
        config_file_path = [];
```

File: plugins/git/lazygit.nix (hunk 4 above)
Comment (theabm) [reply plugins/git/lazygit.nix:26]: I'm not sure. The source code only uses it as a multiplicative factor: look [here](https://github.com/kdheepak/lazygit.nvim/blob/0ada6c6e7e138df92f5009b6952f4ac41248305a/lua/lazygit/window.lua#L19)

So I thought there were no limitations as long as it was non negative.

Comment (GaetanLepage) [reply plugins/git/lazygit.nix:26]: Ok, let's keep it this way then.

File: tests/test-sources/plugins/git/lazygit.nix (hunk 6 above)
Comment (theabm) [reply tests/test-sources/plugins/git/lazygit.nix]: done (and I also modified the settingsExample for coherency)

File: plugins/git/lazygit.nix (hunk 5 above)
Comment (theabm) [reply plugins/git/lazygit.nix]: done

**Review by GaetanLepage — APPROVED** [review]
LGTM ! Good job :)

**Pull request merged**